heat/classification/tests/test_knn.py

```diff
@@ -10,14 +10,9 @@ class TestKNN(TestCase):
     def test_split_none(self):
         x = ht.load_hdf5("heat/datasets/iris.h5", dataset="data")
 
-        # generate keys for the iris.h5 dataset
-        keys = []
-        for i in range(50):
-            keys.append(0)
-        for i in range(50, 100):
-            keys.append(1)
-        for i in range(100, 150):
-            keys.append(2)
+        keys = [0 for _ in range(50)]
+        keys.extend(1 for _ in range(50, 100))
+        keys.extend(2 for _ in range(100, 150))
```

Comment:
Thanks for spotting this. It's actually a very weird way of setting values in an array.

I'd replace the whole thing with with:
```python
keys = ht.zeros(150)
keys[50:100] = 1
keys[100:] = 2
```
What do you think?

Reply:
This looks good...👍. More readable too. I will change all of these in this file to the above format.

Reply:
@ClaudiaComito mam, 2 errors are coming in the tests, and both are from this (`test_knn.py`) file only...Here by using `ht.zeros(150)` we are changing the dtype of "keys, and y = ht.array(keys)" from `ht.int64 to ht.float32`. So, I think that might be the reason but I'm not sure. Shall I change the code accordingly and try?

Reply:
@Sai-Suraj-27 sorry I only saw this now. Quite possible that the dtype change created the problem. You can set `labels = ht.zeros(150, dtype=ht.int32)`. Same applies below.

Reply:
@ClaudiaComito mam, you mean `keys = ht.zeros(150, dtype=ht.int32)` ? because `labels = ht.array(keys, split=0)`. Can you confirm once?

Reply:
@Sai-Suraj-27 I understand the confusion now. Basically I don't think we need to go through `keys` at all. We create the `labels` array directly.

```python
labels = ht.zeros(150, dtype=ht.int32)
labels[50:100] = 1
labels[100:] = 2
```

Reply:
@ClaudiaComito mam, I made the suggested changes, it was slightly confusing because there was no `labels` variable in the functions, 1. `test_split_none`, and 2. `test_split_zero`, it was previously present only in the function `test_fit_one_hot`. I made changes as per what I understood, Please check the changes made in this file before triggering the CI, and tell me if there is anything else to be changed, Thank you.